code/modules/mob/living/simple_animal/animals/fish.dm

```diff
@@ -0,0 +1,73 @@
+// Different types of fish! They are all subtypes of this tho
+/mob/living/simple_animal/fish
+	name = "fish"
+	desc = "Its a fishy.  No touchy fishy."
+	icon = 'icons/mob/fish.dmi'
+	meat_type = /obj/item/weapon/reagent_containers/food/snacks
+
+	// By defautl they can be in any water turf.  Subtypes might restrict to deep/shallow etc
+	var/global/list/suitable_turf_types =  list(
+		/turf/simulated/floor/beach/water,
+		/turf/simulated/floor/beach/coastline,
+		/turf/simulated/floor/holofloor/beach/water,
+		/turf/simulated/floor/holofloor/beach/coastline,
+		/turf/simulated/floor/water
+	)
+
+// Don't swim out of the water
+/mob/living/simple_animal/fish/handle_wander_movement()
+	if(isturf(src.loc) && !resting && !buckled && canmove) //Physically capable of moving?
+		lifes_since_move++ //Increment turns since move (turns are life() cycles)
+		if(lifes_since_move >= turns_per_move)
+			if(!(stop_when_pulled && pulledby)) //Some animals don't move when pulled
+				var/moving_to = 0 // otherwise it always picks 4, fuck if I know.   Did I mention fuck BYOND
+				moving_to = pick(cardinal)
+				dir = moving_to			//How about we turn them the direction they are moving, yay.
+				var/turf/T = get_step(src,moving_to)
+				if(T && is_type_in_list(T, suitable_turf_types))
+					Move(T)
+					lifes_since_move = 0
+
+// Take damage if we are not in water
+/mob/living/simple_animal/fish/handle_breathing()
+	var/turf/T = get_turf(src)
+	if(T && !is_type_in_list(T, suitable_turf_types))
+		if(prob(50))
+			say(pick("Blub", "Glub", "Burble"))
```

Comment:
Yes, but only when the fish is *out of the water*.  When in water they say nothing.